diff -r fdb45920dfe9 src/materials/Textured.gs
--- a/src/materials/Textured.gs	Sun Dec 09 21:17:29 2012 -0500
+++ b/src/materials/Textured.gs	Sun Dec 09 21:23:18 2012 -0500
@@ -23,6 +23,13 @@
 
 
 class soy.materials.Textured : soy.materials.Colored
+    /*
+    *
+    * GL_TEXTURE0 : for colormap
+    * GL_TEXTURE1 : for glowmap
+    * GL_TEXTURE2 : for bumpmap
+    */
+
     // This simplifies rendering code
     _ids : static array of GLenum = {
         GL_TEXTURE0,
@@ -69,6 +76,8 @@
             // Enable colormap if available
             if _colormap is not null
                 self.enable_colormap(0)
+            if _glowmap is not null
+                self.enable_glowmap(1)
 
             // Smooth or flat shading
             self.enable_shades()
@@ -78,14 +87,22 @@
             glDisableClientState(GL_TEXTURE_COORD_ARRAY)
             glClientActiveTexture(GL_TEXTURE2)
             glDisableClientState(GL_TEXTURE_COORD_ARRAY)
-            glClientActiveTexture(GL_TEXTURE1)
-            glDisableClientState(GL_TEXTURE_COORD_ARRAY)
+            if _glowmap is null            
+                glClientActiveTexture(GL_TEXTURE1)
+                glDisableClientState(GL_TEXTURE_COORD_ARRAY)
+            if _colormap is null
+                glClientActiveTexture(GL_TEXTURE0)
+                glDisableClientState(GL_TEXTURE_COORD_ARRAY)
+
             return true
 
         // Cleanup on pass 1
         if _colormap is not null
             self.disable_colormap(0)
 
+        if _glowmap is not null
+            self.disable_glowmap(1)
+
         // Lock mutex so the texture can't be changed until we're done
         _mutex.unlock()
         return false
@@ -96,7 +113,7 @@
         glActiveTexture(_ids[id])
         glClientActiveTexture(_ids[id])
         glEnableClientState(GL_TEXTURE_COORD_ARRAY)
-        glTexCoordPointer(3, GL_FLOAT, (GLsizei) (sizeof(GLfloat) * 11),
+        glTexCoordPointer(3, GL_FLOAT, (GLsizei) (sizeof(GLfloat) * 9),
                           (GLvoid*) (sizeof(GLfloat) * 6))
         // glTexEnvi(GL_TEXTURE_ENV, GL_TEXTURE_ENV_MODE, GL_MODULATE)
         _bumpmap.enable()
@@ -107,7 +124,7 @@
         glActiveTexture(_ids[id])
         glClientActiveTexture(_ids[id])
         glEnableClientState(GL_TEXTURE_COORD_ARRAY)
-        glTexCoordPointer(3, GL_FLOAT, (GLsizei) (sizeof(GLfloat) * 11),
+        glTexCoordPointer(3, GL_FLOAT, (GLsizei) (sizeof(GLfloat) * 9),
                           (GLvoid*) (sizeof(GLfloat) * 6))
         glTexEnvi(GL_TEXTURE_ENV, GL_TEXTURE_ENV_MODE, (GLint) GL_MODULATE)
         _colormap.enable()
@@ -118,9 +135,11 @@
         glActiveTexture(_ids[id])
         glClientActiveTexture(_ids[id])
         glEnableClientState(GL_TEXTURE_COORD_ARRAY)
-        glTexCoordPointer(3, GL_FLOAT, (GLsizei) (sizeof(GLfloat) * 11),
+        glTexCoordPointer(3, GL_FLOAT, (GLsizei) (sizeof(GLfloat) * 9),
                           (GLvoid*) (sizeof(GLfloat) * 6))
-        // glTexEnvi(GL_TEXTURE_ENV, GL_TEXTURE_ENV_MODE, GL_MODULATE)
+        glTexEnvi(GL_TEXTURE_ENV, GL_TEXTURE_ENV_MODE, (GLint) GL_ADD)
+        glEnable(GL_BLEND)
+        glBlendFunc(GL_ONE, GL_ONE)
         _glowmap.enable()
 
 
